lib/std/array_list.zig

```diff
@@ -355,12 +376,22 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
     };
 }
 
-/// Bring-your-own allocator with every function call.
-/// Initialize directly and deinitialize with `deinit` or use `toOwnedSlice`.
+/// An `ArrayList`, but with bring-your-own allocator on every function call.
+/// Initialize directly or with `initCapacity`, and deinitialize with `deinit`
+/// or use `toOwnedSlice`.
+///
+/// Generally you should prefer to use the analogous `ArrayList` unless
+/// you have a reason not to.
```

Comment:
I don't agree with this advice. The API usage of ArrayListUnmanaged can sometimes actually be nicer than ArrayList.

Reply:
I agree with @andrewrk. For instance, in stage2 we make excessive use of `ArrayListUnmanaged` and friends which is simply easier to manage given a deep nesting of structs which all require an allocator, but at the same time, you'd like to be able to have them default-constructed without the necessity to pass in an explicit `Allocator` to the constructor (i.e., the `init` function).

Reply:
I'll remove this advice.